components/scream/src/share/atm_process/atmosphere_process_group.cpp

```diff
@@ -144,25 +144,162 @@ void AtmosphereProcessGroup::finalize_impl (/* what inputs? */) {
 }
 
 void AtmosphereProcessGroup::
-set_required_group_impl (const FieldGroup<const Real>& group)
-{
+set_required_field (const Field<const Real>& f) {
+  // Find the first process that requires this group
+  const auto& fid = f.get_header().get_identifier();
+  int first_proc_that_needs_f = -1;
+  for (int iproc=0; iproc<m_group_size; ++iproc) {
+    if (m_atm_processes[iproc]->requires_field(fid)) {
+      first_proc_that_needs_f = iproc;
+      break;
+    }
+  }
+
+  EKAT_REQUIRE_MSG (first_proc_that_needs_f>=0,
+    "Error! This atmosphere process does not require the input field.\n"
+    "    field id: " + f.get_header().get_identifier().get_id_string() + "\n"
+    "   atm process: " + this->name() + "\n"
+    "Something is wrong up the call stack. Please, contact developers.\n");
+
+  // Loop over all the fields in the group, and see if they are all computed
+  // by a previous process. If so, then this group is not a "required" group
```

Comment:
Is there a better way to say this? I think the copious use of the word "group" is a bit confusing.